src/tv2_afvd_showstyle/migrations/util.ts

```diff
@@ -105,3 +107,102 @@ export function getOutputLayerDefaultsMigrationSteps(versionStr: string): Migrat
 		})
 	)
 }
+
+/**
+ * "Renames" the id of a table by copying over all values over into a new table which has the new id - then removes the values from the old table
+ */
+export function renameTableId(version: string, oldTableId: string, newTableId: string): MigrationStepShowStyle {
+	return {
+		id: `${version}.rename.table.id.${oldTableId}.to.${newTableId}`,
+		version,
+		canBeRunAutomatically: true,
+		validate: (context: MigrationContextShowStyle) => {
+			const oldConfigTable = (context.getBaseConfig(oldTableId) as unknown) as TableConfigItemValue
+			if (!oldConfigTable || oldConfigTable.length === 0) {
+				return false
+			}
+
+			const newConfigTable = (context.getBaseConfig(newTableId) as unknown) as TableConfigItemValue
+			return !newConfigTable
+		},
+		migrate: (context: MigrationContextShowStyle) => {
+			const oldConfigTable = (context.getBaseConfig(oldTableId) as unknown) as TableConfigItemValue
+			const newConfigTable = ((context.getBaseConfig(newTableId) as unknown) as TableConfigItemValue) ?? []
+			oldConfigTable.map(value => newConfigTable.push(value))
+
+			context.setBaseConfig(newTableId, newConfigTable)
+			context.removeBaseConfig(oldTableId)
+		}
+	}
+}
+
+/**
+ * "Renames" the name of a Blueprint configuration by overriding the value of the old configuration onto the new configuration - then removes the old configuration
+ */
+export function renameBlueprintConfiguration(
+	version: string,
+	oldConfigurationName: string,
+	newConfigurationName: string
+): MigrationStepShowStyle {
+	return {
+		id: `${version}.rename.blueprint.configuration.${oldConfigurationName}.to.${newConfigurationName}`,
+		version,
+		canBeRunAutomatically: true,
+		validate: (context: MigrationContextShowStyle) => {
+			const oldConfig = (context.getBaseConfig(oldConfigurationName) as unknown) as BasicConfigItemValue
+			if (!oldConfig || Object.keys(oldConfig).length === 0) {
+				return false
+			}
+
+			const newConfig = (context.getBaseConfig(newConfigurationName) as unknown) as BasicConfigItemValue
+			return !newConfig
```

Comment:
The `Object.keys(oldConfig).length === 0` was necessary when I just made the old config empty instead of removing it. After your suggestion to remove it, then this snippet should no longer be necessary :)

Reply:
Updated